### PR TITLE
Add dialog box trigger

### PR DESCRIPTION
**Pull request by RiByryn**
Implemented the run_dialog_box function, which will fill the statestore DB, which will be the trigger for creating a dialog box on the frontend

### CODE REVIEW

File: docs/documentation/pytest_hardpy.md [hunk 1]
```diff
@@ -178,6 +178,51 @@ def test_message():
     set_message("Update message 2", "msg_upd")
 ```
 
+#### run_dialog_box
+
+Displays a dialog box and updates the 'dialog_box' field in the statestore database.
```
Comment (xorialexandrov) [inline docs/documentation/pytest_hardpy.md:183]: **statestore** should be bold.

File: docs/documentation/database.md [hunk 2]
```diff
@@ -288,7 +309,7 @@ Example of a **current** document:
 ### Runstore scheme
 
 The **runstore** database contains all the fields about the **statestore** database
-plus artifact fields for the test run, module, and case.
+plus artifact fields for the test run, module, and case and without dialog box info..
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply docs/documentation/database.md:312]: готово

File: docs/documentation/database.md [hunk 3]
```diff
@@ -278,6 +289,16 @@ Example of a **current** document:
               "msg": [
                 "Current minute 21"
               ],
+              "dialog_box": {
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply docs/documentation/database.md:292]: Done

File: docs/documentation/pytest_hardpy.md (hunk 1 above)
Comment (RiByryn) [reply docs/documentation/pytest_hardpy.md:183]: Done

File: hardpy/pytest_hardpy/db/schema.py [hunk 4]
```diff
@@ -35,6 +35,7 @@ class CaseStateStore(IBaseResult):
 
     assertion_msg: str | None = None
     msg: dict | None = None
+    dialog_box: dict = {}
```
Comment (xorialexandrov) [inline hardpy/pytest_hardpy/db/schema.py:38]: Need an example in docstring here and in `ResultStateStore` docstring

File: docs/documentation/pytest_hardpy.md [hunk 5]
```diff
@@ -178,6 +178,51 @@ def test_message():
     set_message("Update message 2", "msg_upd")
 ```
 
+#### run_dialog_box
+
+Displays a dialog box and updates the 'dialog_box' field in the **statestore** database.
+
+    Args:
```
Comment (xorialexandrov) [inline docs/documentation/pytest_hardpy.md:185]: The argument is `DialogBoxData`. Required to describe what `DialogBoxData` is.

File: hardpy/pytest_hardpy/pytest_call.py [hunk 6]
```diff
@@ -26,6 +27,46 @@ class CurrentTestInfo(object):
...
+        dialog_text (str): dialog text
+        widget_info (DialogBoxWidget | None): widget info
+    """
+
+    title_bar: str
+    dialog_text: str
+    widget_info: DialogBoxWidget | None
+
+
+class DialogBoxResponse:
```
Comment (xorialexandrov) [inline hardpy/pytest_hardpy/pytest_call.py:62]: Why?

File: hardpy/pytest_hardpy/pytest_call.py [hunk 7]
```diff
@@ -26,6 +27,46 @@ class CurrentTestInfo(object):
...
+
+    Args:
+        title_bar (str): title bar
+        dialog_text (str): dialog text
+        widget_info (DialogBoxWidget | None): widget info
+    """
+
+    title_bar: str
+    dialog_text: str
+    widget_info: DialogBoxWidget | None
```
Comment (xorialexandrov) [inline hardpy/pytest_hardpy/pytest_call.py:59]: Need to move in `dialog_box.py` in utils dir.

File: hardpy/pytest_hardpy/pytest_call.py [hunk 8]
```diff
@@ -203,6 +244,56 @@ def set_driver_info(drivers: dict) -> None:
     reporter.update_db_by_doc()
 
 
+def run_dialog_box(data: DialogBoxData):
+    """Displays a dialog box and updates the 'dialog_box' field in the statestore database.
+
+    Args:
```
Comment (xorialexandrov) [inline hardpy/pytest_hardpy/pytest_call.py:250]: Incorrect docstring

File: hardpy/pytest_hardpy/pytest_call.py (hunk 6 above)
Comment (RiByryn) [reply hardpy/pytest_hardpy/pytest_call.py]: This has been removed

File: hardpy/pytest_hardpy/pytest_call.py (hunk 7 above)
Comment (RiByryn) [reply hardpy/pytest_hardpy/pytest_call.py]: Done, but I'm not sure the import is done correctly

File: hardpy/pytest_hardpy/pytest_call.py (hunk 8 above)
Comment (RiByryn) [reply hardpy/pytest_hardpy/pytest_call.py:214]: Ready

File: docs/documentation/pytest_hardpy.md (hunk 5 above)
Comment (RiByryn) [reply docs/documentation/pytest_hardpy.md:185]: Ready

File: hardpy/pytest_hardpy/pytest_call.py (hunk 7 above)
Comment (RiByryn) [reply hardpy/pytest_hardpy/pytest_call.py]: Ready

**Pull request merged**